packages/cohere/src/cohere-chat-language-model.ts

```diff
@@ -47,7 +110,58 @@ export class CohereChatLanguageModel implements LanguageModelV2 {
     return this.config.provider;
   }
 
-  private getArgs({
+  private async extractDocuments(prompt: LanguageModelV2Prompt): Promise<
+    Array<{
+      data: { text: string; title?: string };
+    }>
+  > {
+    const documents: Array<{ data: { text: string; title?: string } }> = [];
+
+    for (const message of prompt) {
+      if (message.role === 'user') {
+        for (const part of message.content) {
+          if (part.type === 'file') {
+            let textContent: string;
+
+            if (typeof part.data === 'string') {
+              // Base64 or text data
+              textContent = part.data;
+            } else if (part.data instanceof Uint8Array) {
+              // For text files, use TextDecoder
+              if (
+                part.mediaType?.startsWith('text/') ||
+                part.mediaType === 'application/json'
+              ) {
+                textContent = new TextDecoder().decode(part.data);
+              } else {
+                // Skip non-text files for Cohere (PDFs, images, etc.)
+                console.warn(
+                  `Skipping non-text file: ${part.filename || 'unknown'} (${part.mediaType})`,
+                );
```

Comment:
we would need to expose proper warnings. the providers should never log to console